### ISSUE TITLE
Business case of my LES cannot be calculated

### ISSUE THREAD

**Issue by DorinevanderVlies** [quintel/etmoses#1530]
In this [LES](https://beta-moses.energytransitionmodel.com/testing_grounds/767) the business case cannot be calculated. 

Is it possible to see what's going wrong? 

![image](https://cloud.githubusercontent.com/assets/19907658/19727208/c5b553c4-9b8e-11e6-837d-6294ab669860.png)

**Comment by grdw** [comment]
Your business case was never calculated. I'm currently figuring out why.

**Comment by grdw** [comment]
I got dragged of a bit into other issues. 😅 But to reply, your topology contains `String` values which is a bit of a problem to the business case calculation.

````json
{
	"name": "HV",
	"stakeholder": "system operator",
	"depth": 0,
	"x": 0,
	"y": 0,
	"id": 1,
	"children": [{
		"name": "Kabel 1",
		"stakeholder": "system operator",
		"depth": 1,
		"x": 0,
		"y": 100,
		"id": 3,
		"children": [{
			"name": "6",
			"stakeholder": "aggregator",
			"depth": 2,
			"x": -555,
			"y": 200,
			"id": 4,
			"investment_cost": "",
			"technical_lifetime": "",
			"capacity": "",
			"children": [{
				"name": "6 - flat",
				"stakeholder": "customer AC1",
				"depth": 3,
				"x": -675,
				"y": 300,
				"id": 24,
				"investment_cost": "",
				"technical_lifetime": "",
				"capacity": ""
			}, {
				"name": "6 - portiek",
				"stakeholder": "customer AC2",
				"depth": 3,
				"x": -595,
				"y": 300,
				"id": 25,
				"investment_cost": "",
				"technical_lifetime": "",
				"capacity": ""
			}, {
				"name": "6 - rijwoning",
				"stakeholder": "customer AC3",
				"depth": 3,
				"x": -515,
				"y": 300,
				"id": 26,
				"investment_cost": "",
				"technical_lifetime": "",
				"capacity": ""
			}, {
				"name": "6 - 2o1k",
				"stakeholder": "customer AC4",
				"depth": 3,
				"x": -435,
				"y": 300,
				"id": 27,
				"investment_cost": "",
				"technical_lifetime": "",
				"capacity": ""
			}]
		}, {
			"name": "Kabel 2",
			"stakeholder": "aggregator",
			"depth": 2,
			"x": 555,
			"y": 200,
			"id": 5,
			"children": [{
				"name": "9",
				"stakeholder": "aggregator",
				"depth": 3,
				"x": -275,
				"y": 300,
				"id": 6,
				"investment_cost": "",
				"technical_lifetime": "",
				"capacity": "",
				"children": [{
					"name": "9 - flat",
					"stakeholder": "customer AC1",
					"depth": 4,
					"x": -395,
					"y": 400,
					"id": 28,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": ""
				}, {
					"name": "9 - portiek",
					"stakeholder": "customer AC2",
					"depth": 4,
					"x": -315,
					"y": 400,
					"id": 29,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": ""
				}, {
					"name": "9 - rijwoning",
					"stakeholder": "customer AC3",
					"depth": 4,
					"x": -235,
					"y": 400,
					"id": 30,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": ""
				}, {
					"name": "9 - 2o1k",
					"stakeholder": "customer AC4",
					"depth": 4,
					"x": -155,
					"y": 400,
					"id": 31,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": ""
				}]
			}, {
				"name": "Kabel 3",
				"stakeholder": "aggregator",
				"depth": 3,
				"x": 205,
				"y": 300,
				"id": 7,
				"children": [{
					"name": "10",
					"stakeholder": "aggregator",
					"depth": 4,
					"x": 5,
					"y": 400,
					"id": 8,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": "",
					"children": [{
						"name": "10 - flat",
						"stakeholder": "customer AC1",
						"depth": 5,
						"x": -115,
						"y": 500,
						"id": 32,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "10 - portiek",
						"stakeholder": "customer AC2",
						"depth": 5,
						"x": -35,
						"y": 500,
						"id": 33,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "10 - rijwoning",
						"stakeholder": "customer AC3",
						"depth": 5,
						"x": 45,
						"y": 500,
						"id": 34,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "10 - 2o1k",
						"stakeholder": "customer AC4",
						"depth": 5,
						"x": 125,
						"y": 500,
						"id": 35,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}]
				}, {
					"name": "12",
					"stakeholder": "aggregator",
					"depth": 4,
					"x": 405,
					"y": 400,
					"id": 23,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": "",
					"children": [{
						"name": "12 - flat",
						"stakeholder": "customer AC1",
						"depth": 5,
						"x": 285,
						"y": 500,
						"id": 36,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "12 - portiek",
						"stakeholder": "customer AC2",
						"depth": 5,
						"x": 365,
						"y": 500,
						"id": 37,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "12 - rijwoning",
						"stakeholder": "customer AC3",
						"depth": 5,
						"x": 445,
						"y": 500,
						"id": 38,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "12 - 2o1k",
						"stakeholder": "customer AC4",
						"depth": 5,
						"x": 525,
						"y": 500,
						"id": 39,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}]
				}],
				"investment_cost": "",
				"technical_lifetime": "",
				"capacity": ""
			}, {
				"name": "Kabel 4",
				"stakeholder": "aggregator",
				"depth": 3,
				"x": 1385,
				"y": 300,
				"id": 9,
				"children": [{
					"name": "7",
					"stakeholder": "aggregator",
					"depth": 4,
					"x": 805,
					"y": 400,
					"id": 10,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": "",
					"children": [{
						"name": "7 - flat",
						"stakeholder": "customer AC1",
						"depth": 5,
						"x": 685,
						"y": 500,
						"id": 40,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "7 - portiek",
						"stakeholder": "customer AC2",
						"depth": 5,
						"x": 765,
						"y": 500,
						"id": 41,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "7 - rijwoning",
						"stakeholder": "customer AC3",
						"depth": 5,
						"x": 845,
						"y": 500,
						"id": 42,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}, {
						"name": "7 - 2o1k",
						"stakeholder": "customer AC4",
						"depth": 5,
						"x": 925,
						"y": 500,
						"id": 43,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": ""
					}]
				}, {
					"name": "Kabel 5",
					"stakeholder": "aggregator",
					"depth": 4,
					"x": 1965,
					"y": 400,
					"id": 11,
					"investment_cost": "",
					"technical_lifetime": "",
					"capacity": "",
					"children": [{
						"name": "4",
						"stakeholder": "aggregator",
						"depth": 5,
						"x": 1085,
						"y": 500,
						"id": 12,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": "",
						"children": [{
							"name": "4 - flat",
							"stakeholder": "customer AC1",
							"depth": 6,
							"x": 965,
							"y": 600,
							"id": 44,
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": ""
						}, {
							"name": "4 - portiek",
							"stakeholder": "customer AC2",
							"depth": 6,
							"x": 1045,
							"y": 600,
							"id": 45,
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": ""
						}, {
							"name": "4 - rijwoning",
							"stakeholder": "customer AC3",
							"depth": 6,
							"x": 1125,
							"y": 600,
							"id": 46,
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": ""
						}, {
							"name": "4 - 2o1k",
							"stakeholder": "customer AC4",
							"depth": 6,
							"x": 1205,
							"y": 600,
							"id": 47,
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": ""
						}]
					}, {
						"name": "Kabel 6",
						"stakeholder": "aggregator",
						"depth": 5,
						"x": 1765,
						"y": 500,
						"id": 14,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": "",
						"children": [{
							"name": "1",
							"stakeholder": "aggregator",
							"depth": 6,
							"x": 1365,
							"y": 600,
							"id": 16,
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": "",
							"children": [{
								"name": "1 - flat",
								"stakeholder": "customer AC1",
								"depth": 7,
								"x": 1245,
								"y": 700,
								"id": 48,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "1 - portiek",
								"stakeholder": "customer AC2",
								"depth": 7,
								"x": 1325,
								"y": 700,
								"id": 49,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "1 - rijwoning",
								"stakeholder": "customer AC3",
								"depth": 7,
								"x": 1405,
								"y": 700,
								"id": 50,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "1 - 2o1k",
								"stakeholder": "customer AC4",
								"depth": 7,
								"x": 1485,
								"y": 700,
								"id": 51,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}]
						}, {
							"name": "2",
							"stakeholder": "aggregator",
							"depth": 6,
							"x": 1765,
							"y": 600,
							"id": 17,
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": "",
							"children": [{
								"name": "2 - flat",
								"stakeholder": "customer AC1",
								"depth": 7,
								"x": 1645,
								"y": 700,
								"id": 52,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "2 - portiek",
								"stakeholder": "customer AC2",
								"depth": 7,
								"x": 1725,
								"y": 700,
								"id": 53,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "2 - rijwoning",
								"stakeholder": "customer AC3",
								"depth": 7,
								"x": 1805,
								"y": 700,
								"id": 54,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "2 - 2o1k",
								"stakeholder": "customer AC4",
								"depth": 7,
								"x": 1885,
								"y": 700,
								"id": 55,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}]
						}, {
							"name": "3",
							"stakeholder": "aggregator",
							"depth": 6,
							"x": 2165,
							"y": 600,
							"id": 18,
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": "",
							"children": [{
								"name": "3 - flat",
								"stakeholder": "customer AC1",
								"depth": 7,
								"x": 2045,
								"y": 700,
								"id": 56,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "3 - portiek",
								"stakeholder": "customer AC2",
								"depth": 7,
								"x": 2125,
								"y": 700,
								"id": 57,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "3 - rijwoning",
								"stakeholder": "customer AC3",
								"depth": 7,
								"x": 2205,
								"y": 700,
								"id": 58,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}, {
								"name": "3 - 2o1k",
								"stakeholder": "customer AC4",
								"depth": 7,
								"x": 2285,
								"y": 700,
								"id": 59,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": ""
							}]
						}]
					}, {
						"name": "Kabel 7",
						"stakeholder": "aggregator",
						"depth": 5,
						"x": 2845,
						"y": 500,
						"id": 15,
						"investment_cost": "",
						"technical_lifetime": "",
						"capacity": "",
						"children": [{
							"name": "Kabel 8",
							"stakeholder": "aggregator",
							"depth": 6,
							"x": 2845,
							"y": 600,
							"id": 19,
							"children": [{
								"name": "5",
								"stakeholder": "aggregator",
								"depth": 7,
								"x": 2445,
								"y": 700,
								"id": 20,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": "",
								"children": [{
									"name": "5 - flat",
									"stakeholder": "customer AC1",
									"depth": 8,
									"x": 2325,
									"y": 800,
									"id": 60,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "5 - portiek",
									"stakeholder": "customer AC2",
									"depth": 8,
									"x": 2405,
									"y": 800,
									"id": 61,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "5 - rijwoning",
									"stakeholder": "customer AC3",
									"depth": 8,
									"x": 2485,
									"y": 800,
									"id": 62,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "5 - 2o1k",
									"stakeholder": "customer AC4",
									"depth": 8,
									"x": 2565,
									"y": 800,
									"id": 63,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}]
							}, {
								"name": "8",
								"stakeholder": "aggregator",
								"depth": 7,
								"x": 2845,
								"y": 700,
								"id": 21,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": "",
								"children": [{
									"name": "8 - flat",
									"stakeholder": "customer AC1",
									"depth": 8,
									"x": 2725,
									"y": 800,
									"id": 64,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "8 - portiek",
									"stakeholder": "customer AC2",
									"depth": 8,
									"x": 2805,
									"y": 800,
									"id": 65,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "8 - rijwoning",
									"stakeholder": "customer AC3",
									"depth": 8,
									"x": 2885,
									"y": 800,
									"id": 66,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "8 - 2o1k",
									"stakeholder": "customer AC4",
									"depth": 8,
									"x": 2965,
									"y": 800,
									"id": 67,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}]
							}, {
								"name": "11",
								"stakeholder": "aggregator",
								"depth": 7,
								"x": 3245,
								"y": 700,
								"id": 22,
								"investment_cost": "",
								"technical_lifetime": "",
								"capacity": "",
								"children": [{
									"name": "11 - flat",
									"stakeholder": "customer AC1",
									"depth": 8,
									"x": 3125,
									"y": 800,
									"id": 68,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "11 - portiek",
									"stakeholder": "customer AC2",
									"depth": 8,
									"x": 3205,
									"y": 800,
									"id": 69,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "11 - rijwoning",
									"stakeholder": "customer AC3",
									"depth": 8,
									"x": 3285,
									"y": 800,
									"id": 70,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}, {
									"name": "11 - 2o1k",
									"stakeholder": "customer AC4",
									"depth": 8,
									"x": 3365,
									"y": 800,
									"id": 71,
									"investment_cost": "",
									"technical_lifetime": "",
									"capacity": ""
								}]
							}],
							"investment_cost": "",
							"technical_lifetime": "",
							"capacity": ""
						}]
					}]
				}],
				"investment_cost": "",
				"technical_lifetime": "",
				"capacity": ""
			}],
			"investment_cost": "",
			"technical_lifetime": "",
			"capacity": ""
		}],
		"investment_cost": "",
		"technical_lifetime": "",
		"capacity": ""
	}],
	"investment_cost": "",
	"technical_lifetime": "",
	"capacity": ""
}
````

This problem will be fixed once #1517 is merged.

**Comment by grdw** [comment]
Some LES's still contain stringed keys. I'll write a migration to fix those (5) topologies.

Affected topologies:

- Paddepoel Noord optie 1 
- Paddepoel Noord optie 2 
- 20161019 - Paddepoel noord v1
- 20161019 - Paddepoel noord v1 - Clone #1 
- 20161019 - Paddepoel noord v1 - Clone #2
